miniai/learner.py

```diff
@@ -50,13 +50,16 @@ def to_cpu(x):
     return x.detach().cpu()
 
 # %% ../nbs/09_learner.ipynb 35
+from torcheval.metrics import Metric, Mean
+
 class MetricsCB(Callback):
-    def __init__(self, *ms, **metrics):
+    def __init__(self,  *ms, device=def_device, **metrics):
         for o in ms: metrics[type(o).__name__] = o
         self.metrics = metrics
+        for m in self.metrics.values(): m.to(device)
         self.all_metrics = copy(metrics)
-        self.all_metrics['loss'] = self.loss = Mean()
-
+        self.all_metrics['loss'] = self.loss = Mean(device='cpu' if 'mps' in device else device)
```

Comment:
MPS does not support doubles, and Mean fails when placed on device as it weights are double.